src/object.d

```diff
@@ -1183,7 +1191,12 @@ class TypeInfo_Pointer : TypeInfo
 
     override size_t getHash(scope const void* p) @trusted const
     {
-        return cast(size_t)*cast(void**)p;
+        // If the target of `p` is n-byte aligned then the
+        // bottom log2(n) bits of `p` will always be 0.
+        // Performing a constant magnitude xor-shift is
+        // inexpensive and will generally pay off.
+        size_t addr = cast(size_t) *cast(const void**) p;
+        return addr ^ (addr >>> 4);
```

Comment:
This changes sits at odds with https://github.com/dlang/druntime/pull/2243/files#diff-a68e58fcf0de5aa198fcaceafe4e8cf9R1186. Would hashing the pointer address produce better distribution?

Reply:
>Would hashing the pointer address produce better distribution

Possibly. But let's remember the only goal in improving the distribution is to increase AA speed by reducing hash collisions and the speed penalty they incur. Rehashing adds to the time of every hashtable operation so if rehashing takes too long it becomes a net detriment. [In a recent D hash table implementation comparison](https://forum.dlang.org/post/zwkpeeduwrxugoevpmoa@forum.dlang.org) `tanya.container.hashtable` with default settings was severely hurt by the rehashing it performs on integer keys.

Reply:
Makes sense. Let's go with your approach.